apps/account-api/src/controllers/v1/keys-v1.controller.ts

```diff
@@ -76,4 +85,56 @@ export class KeysControllerV1 {
       throw new HttpException('Failed to find public keys for the given msaId', HttpStatus.BAD_REQUEST);
     }
   }
+
+  @Get('publicKeyAgreements/getAddKeyPayload')
+  @HttpCode(HttpStatus.OK)
+  @ApiQuery({
+    name: 'newKey',
+    description: 'New public key to be added in hex format',
+    type: 'string',
+    required: true,
+  })
+  @ApiOperation({ summary: 'Get a properly encoded StatefulStorageItemizedSignaturePayloadV2 that can be signed.' })
+  @ApiOkResponse({ description: 'Returned an encoded StatefulStorageItemizedSignaturePayloadV2 for signing' })
+  /**
+   * Using the provided query parameters, creates a new payload that can be signed to add new graph keys.
+   * @param queryParams - The query parameters for adding a new key
+   * @returns Payload is included for convenience. Encoded payload to be used when signing the transaction.
+   * @throws An error if the key already exists or the payload creation fails.
+   */
+  async getPublicKeyAgreementsKeyPayload(
+    @Query('msaId') msaId: string,
+    @Query('newKey') newKey: HexString,
+  ): Promise<AddNewPublicKeyAgreementPayloadRequest> {
+    // this is temporary until we find a better way to enforce data validation. the validation decorator didn't work
+    if (!isHexString(newKey)) {
+      throw new BadRequestException('Not a valid Hex value!');
+    }
```

Comment:
Validation decorators can work on query parameters if you create a DTO for the `@Query()` parameter and decorate it appropriately. Something like:

```
class AddNewKeyPayloadRequestDto {
  @IsNumberString()
  @ApiProperty({ type: String, description: 'MSA ID of account to generate a payload for' })
  msaId: string;

  @IsHexadecimal()
  @ApiProperty({ type: String, description: 'New public key to be added in hex format' })
  newKey: HexString;
}
```

Then in your controller method signature, instead of individual query parameters, you have a single parameter like so:
```
  async getPublicKeyAgreementsKeyPayload(
    @Query() { msaId, newKey }: AddNewKeyPayloadRequestDto,
  ): Promise<AddNewPublicKeyAgreementPayloadRequest> {

```